From: Luca Coelho <luciano.coelho@intel.com>
Date: Thu, 15 Aug 2019 14:49:00 +0300
Subject: [NOUPSTREAM] backport: rename ktime_get_boot_ns for v5.3+

In v5.3, this function was renamed, so rename it here for v5.3+.  When
we merge v5.3 back from upstream, the opposite should be done (i.e. we
will have _boottime_ and need to rename to _boot_ in < v5.3 instead).

This may be applied to older cores if we want them to compile on
v5.3+.

type=maint
ticket=none

Change-Id: Ia3f2b514a173edb320b3445a9d6e7e31e9182059
Signed-off-by: Luca Coelho <luciano.coelho@intel.com>
Reviewed-on: https://git-amr-3.devtools.intel.com/gerrit/233392
Tested-by: ec ger unix iil jenkins <EC.GER.UNIX.IIL.JENKINS@INTEL.COM>
x-iwlwifi-stack-dev: eb274c49a30f3bf2cf3ca6f537a018707edc0bae
(cherry picked from commit f47729b94e118ad592870073b251f6927b95d1e0)
Signed-off-by: You-Sheng Yang <vicamo@gmail.com>
---
 backport-include/linux/timekeeping.h | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/backport-include/linux/timekeeping.h b/backport-include/linux/timekeeping.h
index 198b09b..b97b09f 100644
--- a/backport-include/linux/timekeeping.h
+++ b/backport-include/linux/timekeeping.h
@@ -26,6 +26,16 @@ static inline u64 ktime_get_boot_ns(void)
 }
 #endif /* < 3.17 */
 
+#if LINUX_VERSION_IS_GEQ(5,3,0)
+/*
+ * In v5.3, this function was renamed, so rename it here for v5.3+.
+ * When we merge v5.3 back from upstream, the opposite should be done
+ * (i.e. we will have _boottime_ and need to rename to _boot_ in <
+ * v5.3 instead).
+*/
+#define ktime_get_boot_ns ktime_get_boottime_ns
+#endif /* > 5.3.0 */
+
 #if LINUX_VERSION_IS_LESS(4,18,0)
 extern time64_t ktime_get_boottime_seconds(void);
 #endif /* < 4.18 */
